packages/client/lib/sync/fullsync.ts

```diff
@@ -8,6 +8,11 @@ import { VMExecution } from './execution/vmexecution'
 import { TxPool } from './txpool'
 import { Event } from '../types'
 
+interface sentBlock {
+  hash: string
```

Comment:
maybe store this as a `Buffer`? since that is our typical internal value for `hash`

Reply:
`sentBlock` -> `SentBlock`

Reply:
@ryanio: just as some note: I had taken the [same string-based datastructure](https://github.com/ethereumjs/ethereumjs-monorepo/pull/1441/files#diff-257377ef4396527158bfb57c5d9b985465a25b021e4e0083de5af8460ee446abR24) on the tx pool since there it was easier to chain on requests  since the hash is used also as string as a key for other data structure.

Not sure if we want to deviate here, seems rather getting a bit inconsistent to me. That said: I am far from sure that I've introduced the optimal data structures along the tx pool PR. So if we generally want to improve here, we can very well do.

Reply:
I switched these to the `buffer.equals...` approach suggested by @ryanio Happy to have it either way but I'm all for any performance optimizations we can do so would tend to go with `buffer.equals` over converting everything to strings if that's more performant.